src/main/java/com/gromoks/filestatistic/service/statistic/FileStatistic.java

```diff
@@ -0,0 +1,124 @@
+package com.gromoks.filestatistic.service.statistic;
+
+import com.gromoks.filestatistic.entity.FileInfo;
+import com.gromoks.filestatistic.entity.LineInfo;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class FileStatistic {
+    public static List<LineInfo> getLinesStatistic(InputStream lines, String wordDelimiter) {
+        List<LineInfo> lineInfos = new ArrayList<>();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(lines))) {
+            String line;
+            int lineNumber = 0;
+
+            while ((line = bufferedReader.readLine()) != null) {
+                lineNumber++;
+                LineInfo lineInfo = FileStatistic.getLineStatistic(line, lineNumber, wordDelimiter);
+                lineInfos.add(lineInfo);
+                System.out.println(lineInfo);
+            }
+        } catch (IOException e) {
+            System.out.println("Error occurred while trying to read the file");
+        }
+
+        return lineInfos;
+    }
+
+    public static FileInfo getAggregateLineStatistic(String fileName, List<LineInfo> lineInfos) {
+        FileInfo fileInfo = new FileInfo();
+        int index = 0;
+        int length = 0;
+        double averageSum = 0;
+        List<String> longestWords = new ArrayList<>();
+        List<String> shortestWords = new ArrayList<>();
+
+        for (LineInfo lineInfo : lineInfos) {
+            length += lineInfo.getRowLength();
+            averageSum += lineInfo.getAverageWordLength();
+            index++;
+
+            if (index == 1) {
+                longestWords = lineInfo.getLongestWords();
+                shortestWords = lineInfo.getShortestWords();
+            } else {
+                if (lineInfo.getLongestWords().get(0).length() > longestWords.get(0).length()) {
+                    longestWords = lineInfo.getLongestWords();
+                } else if (lineInfo.getLongestWords().get(0).length() == longestWords.get(0).length()) {
+                    longestWords.addAll(lineInfo.getLongestWords());
+                }
+
+                if (lineInfo.getShortestWords().get(0).length() < shortestWords.get(0).length()) {
+                    shortestWords = lineInfo.getShortestWords();
+                } else if (lineInfo.getShortestWords().get(0).length() == shortestWords.get(0).length()) {
+                    shortestWords.addAll(lineInfo.getShortestWords());
+                }
```

Comment:
(Optional) I would suggest to store length in intermediate variables to improve code readability